llvm/tools/llvm-shlib/CMakeLists.txt

```diff
@@ -60,8 +53,6 @@ if(LLVM_BUILD_LLVM_DYLIB)
       # inside and outside libLLVM.so.
       target_link_options(LLVM PRIVATE LINKER:-Bsymbolic-functions)
     endif()
-  elseif("${CMAKE_SYSTEM_NAME}" STREQUAL "Darwin")
-    set(LIB_NAMES -Wl,-all_load ${LIB_NAMES})
   endif()
```

Comment:
I think we can generalize `message(FATAL_ERROR "Generating libLLVM is not supported on z/OS")` to all non-ELF-non-MachO OSes and place the fatal error in the `else` branch.

Reply:
Well yes, at the moment if you're not Apple or something ELF then it's not supported. There is AIX, and I guess I thought z/OS support was further along but I see not. But I don't understand how that would work here with CMake.

Reply:
A few lines above we have two checks for MSVC and z/OS. They could be combined into an "else" branch here. Feel free not to do the cleanup though:)

Reply:
Ok, that is what I thought you were getting at, but wanted clarification. I understand that. But the whole point was to eliminate the long list of OS's and one which will only keep growing and making it the fallback path.

I don't see any obvious way of detecting that these are ELF OS's with CMake.